File: 0167-two-sum-ii---input-array-is-sorted/solution.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        k = len(numbers) - 1
        j = 0
        if k==-1:
            return k
        while j<k:
            sum = numbers[j] + numbers[k]
            if sum==target:
                return [j+1, k+1]
            elif sum>target:
                k -= 1
            else:
                j+=1
        return -1

File: 0167-two-sum-ii---input-array-is-sorted/solution.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        Dictionary<int, int> numIndex = new Dictionary<int, int>();
        for(int i=0; i<numbers.Length; i++){
            if (numIndex.ContainsKey(target-numbers[i])){
                return [numIndex[target-numbers[i]]+1, i+1];
            }
            else{
                numIndex[numbers[i]] = i;
            }
        }
        return [];
    }
}
